Ponca/src/SpatialPartitioning/KdTree/kdTree.hpp

```diff
@@ -95,14 +72,15 @@ bool KdTreeBase<Traits>::valid() const
 }
 
 template<typename Traits>
-std::string KdTreeBase<Traits>::to_string(bool verbose) const
+std::string KdTreeImplBase<Traits>::to_string(bool verbose) const
 {
     std::stringstream str;
 
     str << "KdTree:";
     str << "\n  MaxNodes: " << MAX_NODE_COUNT;
     str << "\n  MaxPoints: " << MAX_POINT_COUNT;
-    str << "\n  MaxDepth: " << Traits::MAX_DEPTH;
+    str << "\n  MaxDepth: " << MAX_DEPTH;
+    str << "\n  SupportsSubsampling: " << SUPPORTS_SUBSAMPLING ? "true" : "false";
```

Comment:
See codacy comment:
```suggestion
    str << "\n  SupportsSubsampling: " << (SUPPORTS_SUBSAMPLING ? "true" : "false");
```

Reply:
I remove this since it didn't work when `SUPPORTS_SUBSAMPLING` was redefined in a derived class

Reply:
Making `SUPPORTS_SUBSAMPLING` a template argument of `KdTreeImplBase` would also mean we couldn't use `KdTreeImplBase&` for arguments that can take either a dense or a sparse kdtree (using a template for the kdtree type, e.g. `template <typename KdTree> void f(KdTree& tree) {}` would fix this, but I though it would be too many templates).

Reply:
Fine, let's keep it simple.